include/sys/arc.h

```diff
@@ -110,20 +121,27 @@ typedef enum arc_flags
 	ARC_FLAG_L2_WRITING		= 1 << 11,	/* write in progress */
 	ARC_FLAG_L2_EVICTED		= 1 << 12,	/* evicted during I/O */
 	ARC_FLAG_L2_WRITE_HEAD		= 1 << 13,	/* head of write list */
+	/*
+	 * Encrypted or authenticated on disk (may be plaintext in memory).
+	 * This header has b_crypt_hdr allocated.
+	 */
+	ARC_FLAG_PROTECTED		= 1 << 14,
```

Comment:
The naming here is a little tricky. The 3 general categories of BPs in an encrypted dataset (as far as encryption is concerned) are (1) encrypted bps, (2) authenticated bps, and (3) indirect blocks that maintain a checksum of MACs. They all have to do with cryptography and all use cryptographic hashes and such, but `BP_IS_PROTECTED()` and `ARC_FLAG_PROTECTED` specifically refer to (1) and (2) only. Let me know if you still have a strong opinion either way, but personally I like the current naming.